javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,40 @@
+//One by one
+function translateOneByOne() {
+  const promise1 = new Promise(resolve => {
+    setTimeout(() => resolve(moveElement(document.querySelector("ul.marks li:nth-child(1)"), { x: 20, y: 300 })), 1000)
+  })
+    .then(() => {      
+      console.log("Red circle element has been moved");
+    });
+  const promise2 = new Promise(resolve => {
+    setTimeout(() => resolve(moveElement(document.querySelector("ul.marks li:nth-child(2)"), { x: 400, y: 300 })), 3000)
+  })
+    .then(() => {
+      console.log("Blue circle element has been moved");
+    });
+  const promise3 = new Promise(resolve => {
+    setTimeout(() => resolve(moveElement(document.querySelector("ul.marks li:nth-child(3)"), { x: 400, y: 20 })), 5000)
+  })
+    .then(() => {
+      console.log("Green circle element has been moved");
+    });
+}
```

Comment:
No need to create `promise1`, `promise2` and `promise3` variables
and `moveElement` is a function that returns a promise so you don't have to wrap it with a promise
The function could be like this
```js
function translateOneByOne() {
  moveElement(red, { x: 20, y: 300 })
    .then(() => {
      console.log('red moved');
      return moveElement(blue, { x: 400, y: 300 });
    })
    .then(() => {
      console.log('blue moved');
      return moveElement(green, { x: 400, y: 20 });
    })
    .then(() => {
      console.log('green moved');
    });
}
```

Reply:
hi @zuhair.. Tq for ur review, i understand now, but i dont know that i can use return in  .then(), here i can see its working, but does it only work to return promises? because it didnt work in other cases...